### ISSUE TITLE
Npm packages with dependencies that have version ranges produce really bad pom.xml metadata

### ISSUE THREAD

**Issue by dsyer** [webjars/webjars#1951]
You simply can't trust that any library with a pom.xml that contains a dependency with a version range is ever going to work in practice. It is *hugely* inefficient (Maven has to look at all the candidates and download them), so it can take all day to simply list the dependencies. There are also often errors in the NPM metadata, or inconsistencies with the webjars repositories that actually just break the build.

Here's a sample the fails with `mvn dependency:list`:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
    <modelVersion>4.0.0</modelVersion>

    <packaging>jar</packaging>
    <groupId>com.example</groupId>
    <artifactId>jquery-issue</artifactId>
    <version>0.0.1-SNAPSHOT</version>

    <properties>
        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
    </properties>

    <dependencies>
        
        <dependency>
            <groupId>org.webjars.npm</groupId>
            <artifactId>jquery-ui</artifactId>
            <version>1.13.0</version>
        </dependency>
        
        
    </dependencies>

</project>
```

The problem stems from the fact that `jquery-ui` has this in its `package.json`:

```json
	"dependencies": {
		"jquery": ">=1.8.0 <4.0.0"
	},
```

which translates into this in the webjars `pom.xml`:

```xml
    <dependencies>
        
        <dependency>
            <groupId>org.webjars.npm</groupId>
            <artifactId>jquery</artifactId>
            <version>[1.8.0,4.0.0)</version>
        </dependency>
        
        
    </dependencies>
```

That version range is evil. Version ranges make sense (marginally) for apps. they never work out well for libraries. It would be better to simply depend on the latest version available.

**Comment by jamesward** [comment]
Yeah, Maven version ranges are a bad idea. I think the alternative (not replicating that information) is worse.  But I need to noodle that more.  In the meantime, the issue you ran into was because neither `org.webjars.npm navigator 1.0.0` or `org.webjars.npm navigator 1.0.1` exists.  When an NPM WebJar is deployed the transitive dependencies are also deployed.  But sometimes deployment of those fails.  In the case of `org.webjars.npm navigator 1.0.x` there is an error deploying it:
```
The license could not be determined for NPM - navigator 1.0.1
```
I'll investigate that and see if we can get that dependency deployed.

FWIW, Users do occasionally run into these kinds of issues with missing transitives and removing ranges wouldn't fix that.

**Comment by jamesward** [comment]
Some further info on the issue with `navigator`:
There is no license info: https://github.com/coolaj86/node-navigator
But it is odd that Maven is picking jquery `1.8.3` in the range `[1.8.0,4.0.0)`.  Newer versions of `jquery` don't have the `navigator` dependency.
I'm diving into how Maven resolves the range...

**Comment by jamesward** [comment]
I can't find why Maven is picking the lowest in the range. Seems strange to me.  But I was able to deploy `org.webjars.npm:navigator:1.0.1` so this particular transitive should be fixed.

I'll continue thinking about not using ranges and instead locking to the latest available in the specified range.

**Comment by dsyer** [comment]
Maven tries the resolve the 1.8.3 dependencies because with literally has to download all the poms for all the versions in that (really large) range. It takes about 10 minutes just to fail for me, having worked its way up from 1.8.0. It’s not a good thing.

**Comment by jamesward** [comment]
Side note, Gradle does the *right* thing (doesn't download the universe) and resolves the transitive dependency to be `jquery` `3.6.0`. I need to look further into Maven's range resolution stuff.
